Sources/Conditions/CreditCardCondition.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  CreditCardCondition.swift
+//  FormValidatorSwift
+//
+//  Created by Onur Ersel on 2016-11-02.
+//  Copyright © 2016 ustwo. All rights reserved.
+//
+
+import Foundation
+
+
+/// Credit card types that are supported by `CreditCardCondition`. There is a `regex` property that returns the regex for validating that credit card number.
+public enum CreditCardType : String {
+    case americanExpress    = "American Express"
+    case dinersClub         = "Diners Club"
+    case discover           = "Discover"
+    case jcb                = "JCB"
+    case maestro            = "Maestro"
+    case mastercard         = "Master Card"
+    case visa               = "VISA"
+    case all                = "all"
+    
+    var regex: String {
+        switch self {
+        case .americanExpress:
+            return "3[47][0-9]{13}"
+        case .dinersClub:
+            return "3(?:0[0-5]|[68][0-9])[0-9]{11}"
+        case .discover:
+            return "6(?:011|5[0-9]{2})(?:[0-9]{12})"
+        case .jcb:
+            return "(?:2131|1800|35\\d{3})\\d{11}"
+        case .maestro:
+            return "(?:5[0678]\\d\\d|6304|6390|67\\d\\d)\\d{8,15}"
+        case .mastercard:
+            return "5[1-5][0-9]{14}"
+        case .visa:
+            return "^4[0-9]{12}(?:[0-9]{3})?$"
+        case .all:
```

Comment:
Rather than having an "all" case in the enum, perhaps it would be better to create this as an OptionSet that can generate the appropriate regex and the like from the selected options?

See https://developer.apple.com/reference/swift/optionset